modular_skyrat/master_files/code/modules/pain_shock/status_indicator.dm

```diff
@@ -0,0 +1,198 @@
+		return TRUE
+
+/mob/living/proc/is_paralyzed()
+	if(HAS_TRAIT_FROM(src, TRAIT_FLOORED, CHOKEHOLD_TRAIT) || HAS_TRAIT(src, TRAIT_CRITICAL_CONDITION) || HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, STAMINA) || HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) && HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) && HAS_TRAIT_FROM(src, TRAIT_FLOORED, TRAIT_STATUS_EFFECT(STAT_TRAIT)))
+		return TRUE
+
+/mob/living/proc/is_unconcious()
+	if(HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+		return TRUE
+
+/mob/living/proc/is_confused()
+	if(has_status_effect(/datum/status_effect/confusion))
+		return TRUE
+
+/* /mob/living/carbon/proc/is_blind_status()
+	if(HAS_TRAIT_FROM(src, TRAIT_BLIND, STATUS_EFFECT_TRAIT) || HAS_TRAIT_FROM(src, TRAIT_BLIND, EYES_COVERED) && !HAS_TRAIT_FROM(src, TRAIT_BLIND, QUIRK_TRAIT)) // We don't want permanently blind users (those who took the trait) to have this forever.
+		return TRUE */ // Disabled, uncomment if we want this.
+
+/mob/living/carbon/death(gibbed) // On death, we clear the indiciators
+	..() // Call the TG death. Do not . = ..()!
+	for(var/iteration in status_indicators) // When we die, clear the indicators.
+		remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
+/* /datum/status_effect/incapacitating/on_apply()
+	. = ..()
+	addtimer(CALLBACK(owner, /mob/living/.proc/status_sanity), 100 MILLISECONDS)
+/datum/status_effect/incapacitating/on_remove()
+	. = ..()
+	addtimer(CALLBACK(owner, /mob/living/.proc/status_sanity), 1000 MILLISECONDS) */
+
+/* /mob/living/carbon/apply_effect()
+	. = ..()
+	status_sanity() */
+
+/mob/living/carbon/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_CARBON_HEALTH_UPDATE, .proc/status_sanity)
+
+/* /mob/living/updatehealth()
+	. = ..()
+	status_sanity() */
+/mob/living/proc/status_sanity()
+	SIGNAL_HANDLER
+	if(stat == DEAD)
+		for(var/iteration in status_indicators) // When we die, clear the indicators.
+			remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
+	is_weakened() ? add_status_indicator("weakened") : remove_status_indicator("weakened")
+	is_paralyzed() ? add_status_indicator("paralysis") : remove_status_indicator("paralysis")
+	is_stunned() ? add_status_indicator("stunned") : remove_status_indicator("stunned")
+	is_unconcious() ? add_status_indicator("sleeping") : remove_status_indicator("sleeping")
+	is_confused() ? add_status_indicator("confused") : remove_status_indicator("confused")
+//	is_blind_status() ? add_status_indicator("blinded") : remove_status_indicator("blinded") // Disabled, keeping for IF we decide we want this.
+//	is_critical() ? add_status_indicator("critical") : remove_status_indicator("critical") // Also disabled. Crosses medhud territory, despite being a shift click examine thing.
+
+/mob/living/proc/add_status_indicator(image/thing)
```

Comment:
Document this, please.

Also, as I stated earlier, I think this should have two arguments, the second being an `indicator_source`, so that when you add an indicator, you can specify its source and not duplicate it if you already have one, and instead just add to the list of its sources, and then you can delete entries from the list of sources of said indicator if it exists, and then just remove the status indicator if its list of sources reaches a length of zero, like we can do for traits.